server/src/main/java/org/elasticsearch/index/shard/InternalIndexingStats.java

```diff
@@ -112,18 +112,20 @@ static class StatsHolder {
         private final CounterMetric deleteCurrent = new CounterMetric();
         private final CounterMetric noopUpdates = new CounterMetric();
 
-        IndexingStats.Stats stats(boolean isThrottled, long currentThrottleMillis) {
+        IndexingStats.Stats stats(boolean isThrottled, long currentThrottleMillis, long timeSinceShardStartedInNanos) {
+            final long totalIndexingTimeInNanos = indexMetric.sum();
             return new IndexingStats.Stats(
                 indexMetric.count(),
-                TimeUnit.NANOSECONDS.toMillis(indexMetric.sum()),
+                TimeUnit.NANOSECONDS.toMillis(totalIndexingTimeInNanos),
                 indexCurrent.count(),
                 indexFailed.count(),
                 deleteMetric.count(),
                 TimeUnit.NANOSECONDS.toMillis(deleteMetric.sum()),
                 deleteCurrent.count(),
                 noopUpdates.count(),
                 isThrottled,
-                TimeUnit.MILLISECONDS.toMillis(currentThrottleMillis)
+                TimeUnit.MILLISECONDS.toMillis(currentThrottleMillis),
+                timeSinceShardStartedInNanos > 0 ? (double) totalIndexingTimeInNanos / timeSinceShardStartedInNanos : 0
```

Comment:
Is there a division by 0 risk here?

Reply:
It's guarded by `timeSinceShardStartedInNanos > 0`

Reply:
:facepalm: